src/main/java/redis/clients/jedis/JedisShardInfo.java

```diff
@@ -31,7 +32,7 @@ public int getPort() {
   public JedisShardInfo(String host) {
     super(Sharded.DEFAULT_WEIGHT);
```

Comment:
Seems like we can call this(URL.create(host)) instead of extracting partial URL values to remove duplication.
Could you please check? :)